zebra-network/src/peer_set/candidate_set.rs

```diff
@@ -102,18 +107,22 @@ use crate::{types::MetaAddr, AddressBook, BoxError, PeerAddrState, Request, Resp
 pub(super) struct CandidateSet<S> {
     pub(super) peer_set: Arc<Mutex<AddressBook>>,
     pub(super) peer_service: S,
+    sleep: Sleep,
 }
 
 impl<S> CandidateSet<S>
 where
     S: Service<Request, Response = Response, Error = BoxError>,
     S::Future: Send + 'static,
 {
+    const PEER_CONNECTION_INTERVAL: Duration = Duration::from_millis(100);
```

Comment:
In the design I suggested, this becomes the minimum peer connection interval.
```suggestion
    /// The minimum time between successive calls to `CandidateSet::next()`.
    ///
    /// ## Security
    ///
    /// Zebra resists distributed denial of service attacks by making sure that new peer connections
    /// are initiated at least `MIN_PEER_CONNECTION_INTERVAL` apart.
    const MIN_PEER_CONNECTION_INTERVAL: Duration = Duration::from_millis(100);
```